codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -261,10 +259,25 @@
     int numActiveStreams();
 
     /**
-     * Gets all streams that are actively in use (i.e. {@code OPEN} or {@code HALF CLOSED}). The returned collection is
-     * sorted by priority.
+     * Provide a means of iterating over the collection of active streams.
+     *
+     * @param visitor The visitor which will visit each active stream.
+     * @return The stream before iteration stopped or {@code null} if iteration went past the end.
+     */
+    Http2Stream forEachActiveStream(ActiveStreamVisitor visitor) throws Http2Exception;
+
+    /**
+     * A visitor that allows iteration over the active streams.
      */
-    Collection<Http2Stream> activeStreams();
+    interface ActiveStreamVisitor {
+        /**
+         * @return <ul>
+         *         <li>{@code true} if the processor wants to continue the loop and handle the entry.</li>
+         *         <li>{@code false} if the processor wants to stop handling headers and abort the loop.</li>
+         *         </ul>
+         */
+        boolean visit(Http2Stream stream) throws Http2Exception;
```

Comment:
So the thought is allow the iterator to abort early in a meaningful way if it needs/wants to.  We should be catching `Throwable` anyways.  Still remove?